commit 9e374938aaac
Author: Munro Mengjue Chiang <mchiang@mozilla.com>
Date:   Wed Sep 27 10:16:46 2017 +0800

    Bug 1402377 - Stop the corresponding track instead of the whole stream when an external device is plugged out. r=jib, r=pehrsons, a=ritu
    
    MozReview-Commit-ID: 7gbB5sgn6xI
    
    --HG--
    extra : source : 140805019de38dbfbb9f4fe363de8d7c8a52d387
    extra : histedit_source : 9f60927d8c7dfba265a60497b305e65eeb28939b
---
 dom/media/MediaManager.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git dom/media/MediaManager.cpp dom/media/MediaManager.cpp
index a06296086a28..6b37fa4206a2 100644
--- dom/media/MediaManager.cpp
+++ dom/media/MediaManager.cpp
@@ -4067,14 +4067,14 @@ GetUserMediaWindowListener::StopRawID(const nsString& removedDeviceID)
       nsString id;
       source->GetAudioDevice()->GetRawId(id);
       if (removedDeviceID.Equals(id)) {
-        source->Stop();
+        source->StopTrack(kAudioTrack);
       }
     }
     if (source->GetVideoDevice()) {
       nsString id;
       source->GetVideoDevice()->GetRawId(id);
       if (removedDeviceID.Equals(id)) {
-        source->Stop();
+        source->StopTrack(kVideoTrack);
       }
     }
   }
